overlord/servicestate/quota_handlers.go

```diff
@@ -595,7 +601,34 @@ func ensureSnapServicesStateForGroup(st *state.State, grp *quota.Group, opts *en
 	return restartSnapServices(st, nil, appsToRestartBySnap, nil)
 }
 
+func ensureGroupIsNotMixed(group string, allGrps map[string]*quota.Group) error {
+	grp, ok := allGrps[group]
+	if ok && len(grp.SubGroups) != 0 && len(grp.Snaps) != 0 {
+		return fmt.Errorf("quota group %q has mixed snaps and sub-groups, which is not supported anymore. please remove it and create it again to make any modifications", group)
```

Comment:
super super duper nitpick, but this message is quite long, maybe

```suggestion
		return fmt.Errorf("quota group %q has mixed snaps and sub-groups, which is no longer supported; removal and re-creation is necessary to modify it", group)
```

Reply:
Thanks for this, the length of the message annoyed me a bit but I had a hard time reducing it while keeping it informational.